source/RialtoGStreamerMSEBaseSink.cpp

```diff
@@ -217,6 +217,12 @@ static void rialto_mse_base_sink_finalize(GObject *object)
 
     priv->~RialtoMSEBaseSinkPrivate();
     GST_CALL_PARENT(G_OBJECT_CLASS, finalize, (object));
+
+    if (!firebolt::rialto::IClientLogControlFactory::createFactory()->createClientLogControl().registerLogHandler(nullptr,
```

Comment:
Why is it needed? What if there would be scenario that only audio-sink would be removed from the pipeline and video would stay? Such scenario is possible in some TrickPlay cases.

Reply:
This is because, for example, ClientController::~ClientController() runs AFTER main() finishes
  and this method attempts to use the log (if configured to do so). However, gstreamer logging has potentially
  been freed at this point if the client has called gst_deinit. And this would cause read and write to freed
  memory.

To improve this code I have added a sink reference count so that the logToGst handler is disabled when the last sink is finalised.